src/main/java/eu/fasten/maven/CheckMojo.java

```diff
@@ -441,12 +434,18 @@ private MavenExtendedRevisionJavaCallGraph buildCallGraph(Artifact artifact, Fil
         outputFile.getParentFile().mkdirs();
 
         try {
-            FileUtils.write(outputFile, cg.toJSON().toString(4), StandardCharsets.UTF_8);
+            writeJSON(cg.toJSON(), outputFile);
```

Comment:
Thanks. This reminds me that we should really introduce some `write` or `serialize` API directly in ExtendedRevisionCallGraph at some point to avoid creating the JSONObject.

Reply:
Actually, that does exist in FASTEN core. It builds a string directly without first building a JSONObject. I'm not sure if it reduces memory usage, though.

Reply:
> Actually, that does exist in FASTEN core.

Ok, I missed it then, or maybe it's more recent than my last pull.

> I'm not sure if it reduces memory usage, though.

Well even if none of the strings are duplicated it does remove quite a lot of new JSONObject and HashMap which can cost a lot in such a volume.